web/src/sessions.rs

```diff
@@ -0,0 +1,115 @@
+use axum::{
+    body::Body,
+    extract::FromRequestParts,
+    http::{self},
+    response::{IntoResponse, Response},
+};
+use cja::{app_state::AppState as _, tower_cookies::Cookies};
+
+use uuid::Uuid;
+
+use crate::AppState;
+
+#[derive(Debug)]
+#[allow(dead_code)]
+pub struct Session {
+    #[allow(clippy::struct_field_names)]
+    pub session_id: Uuid,
+    pub user_id: Option<Uuid>,
+    pub expires_at: chrono::DateTime<chrono::Utc>,
+    pub logged_out_at: Option<chrono::DateTime<chrono::Utc>>,
+    pub updated_at: chrono::DateTime<chrono::Utc>,
+    pub created_at: chrono::DateTime<chrono::Utc>,
+}
+
+impl Session {
+    async fn create(state: &AppState, cookies: &Cookies) -> Result<Self, SessionError> {
```

Comment:
This method creates a Session in the DB, AND sets a browser cookie with its value so the next page load we remember who this user is!

Reply:
Ohhh that's sick! Does that mean the user doesn't have to re-login after leaving the page and coming back?